parser/parser.py

```diff
@@ -1,268 +1,263 @@
-#!/usr/bin/env python2.7
 # -*- coding: utf-8 -*-
 
-import urllib2
+import urllib
 import sys
 import json
 import re
-
-attributepattern = re.compile(r"\<p\>(.*?)\</p\>")
+import codecs
+
+lehrformStrings = {
+    "PS"    : [u"Projekt", u"Seminar"],
+    "V"     : [u"Vorlesung"],
+    "S"     : [u"Seminar"],
+    "BS"    : [u"Seminar"],
+    "SP"    : [u"Seminar", u"Projekt"],
+    "VU"    : [u"Vorlesung", u"Übung"],
+    "VP"    : [u"Vorlesung", u"Projekt"],
+    "MP"    : False
+    #more?
+}
+
+sskStrings = {
+    "Kommunikation"         : u"SSKKO",
+    "Schlüsselkompetenzen"  : u"SSKSK",
+    "Management"            : u"SSKMA",
+    "Recht"                 : u"SSKRE",
+    "Design Thinking"       : u"SSKDT"
+}
+
+vertiefungStrings = {
+    "Internet &amp; Security Technology"                            : u"IST",
+    "Software Architecture &amp; Modeling Technology"               : u"SAMT",
+    "Operating Systems &amp; Information Systems Technology"        : u"OSIS",
+    "Business Process &amp; Enterprise Technology"                  : u"BPET",
+    "Human Computer Interaction &amp; Computer Graphics Technology" : u"HCT"
+}
+
+itseStrings = {
+    "IT-Systems Engineering A" : u"ITSE",
+    "IT-Systems Engineering B" : u"ITSE",
+    "IT-Systems Engineering C" : u"ITSE",
+    "IT-Systems Engineering D" : u"ITSE"
+}
 
 def URLForSemester(semester):
-	"""gibt für ein gewünschtes Semester die entsprechende Liste der Lehrveranstaltungen zurück"""
-	if semester == "now":
-		url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/itse.html"
-	else:
-		year = int(semester.lstrip("wWsS"))
-		if year > 99:
-			print "You have been using this program far too long. Please start coding a new one."
-		else:
-			if semester.startswith("SS"):
-				url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ss_20" + (str(year) if year >= 10 else "0" + str(year)) + ".html"
-			else:
-				if (year < 9):
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_0" + str(year) + "0" + str(year+1) + ".html"
-				elif (year == 9):
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_0" + str(year) + str(year+1) + ".html"
-				else:
-					url = "http://www.hpi.uni-potsdam.de/studium/lehrangebot/lehrangebotsarchiv/lehrangebotsarchiv_ws_" + str(year) + str(year+1) + ".html"
-	print url
-	return url
+    # returns list of courses for semester
```

Comment:
afair ``"""text"""`` is the equivalent to Javadoc and probably better here, even if no one is ever going to read it...